dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -88,53 +97,51 @@ def refute_estimate(self):
-            elif len(new_outcome.shape) == 1:
-                assert len(new_outcome) == num_rows, ("The number of outputs do not match that of the number of outcomes")
-            else:
-                raise Exception("Type Mismatch: The outcome is one dimensional, but the output has the shape:{}".format(new_outcome.shape))
+        new_outcome = self._data['y']
+
+        # if callable(self._outcome_function):
+        #     new_outcome = self._outcome_function(self._data)
         
-        else:
-            new_outcome = np.zeros((num_rows,))
 
-        for index in range(self._num_simulations):
+        # elif type(self._outcome_function) is str:
+        #     new_outcome = self._estimate_dummy_outcome()
 
-            if self._dummy_outcome_type == "permute":
-                if self._random_state is None:
-                    new_outcome = self._data[self._outcome_name].sample(frac=1).values
-                else:
-                    new_outcome = self._data[self._outcome_name].sample(frac=1,
-                                                                random_state=self._random_state).values
-            else:
-                new_outcome += np.random.randn(num_rows)
+        # if type(new_outcome) is pd.Series or \
+        #     type(new_outcome) is pd.DataFrame:
+        #     new_outcome = new_outcome.values
+        
+        # # Check if data types match
+        # assert type(new_outcome) is np.ndarray, ("Only  supports numpy.ndarray as the output")
+        # assert 'float' in new_outcome.dtype.name, ("Only float outcomes are currently supported")
+        
+        # self._scale *= np.std(new_outcome)
+
+        # if len(new_outcome.shape) == 2 and \
+        #     ( new_outcome.shape[0] ==1 or new_outcome.shape[1] ):
+        #     self.logger.warning("Converting the row or column vector to 1D array")
+        #     new_outcome = new_outcome.ravel()
+        #     assert len(new_outcome) == num_rows, ("The number of outputs do not match that of the number of outcomes")
+        # elif len(new_outcome.shape) == 1:
+        #     assert len(new_outcome) == num_rows, ("The number of outputs do not match that of the number of outcomes")
+        # else:
+        #     raise Exception("Type Mismatch: The outcome is one dimensional, but the output has the shape:{}".format(new_outcome.shape))
+
+        
+        pdb.set_trace()
+        for index in range(self._num_simulations):
+            for action, props in self._pipeline:
+                if props['input'] == ['X']:
+                    new_outcome = action( self._data[self._chosen_variables] )
+                elif props['input'] == ['y']:
+                    new_outcome = action( new_outcome )
+                elif props['input'] == ['X','y']:
+                    temp_estimator = action( new_outcome )
+                    new_outcome = temp_estimator( self._data[self._chosen_variables] )   

```

Comment:
lines 132-141 are complex enough that they need inline comments. what is action, what is props etc.